api/src/main/java/io/strimzi/api/kafka/model/KafkaResources.java

```diff
@@ -0,0 +1,187 @@
+     */
+    public static String kafkaPodName(String clusterName, int podNum) {
+        return kafkaStatefulSetName(clusterName) + "-" + podNum;
+    }
+
+    /**
+     * Returns the name of the Entity Operator {@code Deployment} for a {@code Kafka} cluster of the given name.
+     * This {@code Deployment} will only exist if {@code Kafka.spec.entityOperator} is configured in the
+     * {@code Kafka} resource with the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding Entity Operator {@code Deployment}.
+     */
+    public static String entityOperatorDeploymentName(String clusterName) {
+        return clusterName + "-entity-operator";
+    }
+
+    /**
+     * Returns the name of the Cluster CA certificate {@code Secret} for a {@code Kafka} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding Cluster CA certificate {@code Secret}.
+     */
+    public static String clusterCaCertificateSecretName(String clusterName) {
+        return clusterName + "-cluster-ca-cert";
+    }
+
+    /**
+     * Returns the name of the Cluster CA key {@code Secret} for a {@code Kafka} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding Cluster CA key {@code Secret}.
+     */
+    public static String clusterCaKeySecretName(String clusterName) {
+        return clusterName + "-cluster-ca";
+    }
+
+    /**
+     * Returns the name of the Clients CA certificate {@code Secret} for a {@code Kafka} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding Clients CA certificate {@code Secret}.
+     */
+    public static String clientsCaCertificateSecretName(String clusterName) {
+        return clusterName + "-clients-ca-cert";
+    }
+
+    /**
+     * Returns the name of the Clients CA key {@code Secret} for a {@code Kafka} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding Clients CA key {@code Secret}.
+     */
+    public static String clientsCaKeySecretName(String clusterName) {
+        return clusterName + "-clients-ca";
+    }
+
+    /**
+     * Returns the name of the internal bootstrap {@code Service} for a {@code Kafka} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding bootstrap {@code Service}.
+     */
+    public static String internalBootstrapServiceName(String clusterName) {
+        return clusterName + "-kafka-bootstrap";
+    }
+
+    /**
+     * Returns the socket address (<em>&lt;host&gt;</em>:<em>&lt;port&gt;</em>)
```

Comment:
I would avoid the low level concept of socket but using connection instead as it is reflected by the method name. The same for all the other following methods.

Reply:
Yeah, I wasn't really sure about the best terminology. Really "connection" is not an recognised term for the `host:port`. I agree socket address might seem low level, but that's what it is. The only other recognised term I could think of was "endpoint", but that has a kube-specific meaning, so I didn't think that was a great choice either.

Reply:
What if you leave it just as address?

Reply:
Should we keep the method name as `*Connection()` or `*Address()`?

Reply:
Yeah, maybe address is better than socket and connection but I would rename the methods as well as suggested.